tests/functional/utilities/reportHandler.js

```diff
@@ -147,37 +175,59 @@ describe('reportHandler::getCRRStats', function testSuite() {
         });
     });
 
-    after(done => {
-        redisClient.clear(err => {
-            assert.ifError(err, `Expected success, but got error ${err}`);
-            redisClient._client.disconnect();
-            return done();
+    afterEach(done => {
+        async.series({
+            clearRedis: next =>
+                redisClient.clear(next),
+            disconnectRedisPopulator: next =>
+                redisClient.disconnect(next),
+        }, done);
+    });
+
+    it('should disconnect backbeat.metrics client on report completion',
+    done => {
+        const redisChecker = new Redis({ host: 'localhost', port: 6379 });
+        redisChecker.once('error', err => {
+            redisClient.disconnect();
+            done(err);
         });
+        let preCheckCount;
+        async.series({
+            preCheck: next => {
+                redisChecker.client('list', (err, res) => {
+                    assert.ifError(err, `Expect success, but got ${err}`);
+                    const clients = res.split('\n').filter(c => !!c);
+                    preCheckCount = clients.length;
+                    next();
+                });
+            },
+            checkResults: next => {
+                getCRRStats(logger, (err, res) => {
+                    assert.ifError(err,
```

Comment:
For the `assert.ifError`, I believe all other times it has been used in tests it only accepts 1 param (err). I also don't see documentation for the function accepting a 2nd param here: https://nodejs.org/api/assert.html#assert_assert_iferror_value

In any case, if `assert.ifError` finds an error, it will output the error info as a stack trace.